api/tableau_api/views.py

```diff
@@ -8,9 +69,180 @@ class SummaryStatisticDailyStudyView(TableauApiView):
+                 'participant_ids': kwargs.get('participant_ids', ''),
+                 'fields': fields}
+
+        form = ApiQueryForm(data=query)
+        for query_val in kwargs:
+            if query_val not in valid_query_parameters:
+                form.add_error(field=None, error=ValidationError(message="%s is not recognized as an API parameter"
+                                                                         % query_val))
+        if not form.is_valid():
+            return form.errors.get_json_data(), None
+        return None, form.cleaned_data
+
+
+class CsvField(forms.CharField):
+    def clean(self, value):
+        value = super().clean(value)
+        value = value.split(",")
+        if value == [""]:
+            return None
+        return value
+
+
+# TODO delete one of these
+
+#  I think these two are functionally equivalent right now but this one
+#  is more likely to fail loudly if something changes (raise an error it shouldnt rather than not raise one it should)
+#  slightly easier to understand and modify, and is shorter
+class AltMultiErrorMultipleChoiceField(forms.MultipleChoiceField):
+    def validate(self, value):
+        errs = []
+        for val in value:
+            try:
+                super().validate([val])
+            except ValidationError as e:
+                errs.append(e)
+        if errs:
+            raise ValidationError(errs, code='invalid_choice')
+
+
+class MultiErrorMultipleChoiceField(forms.MultipleChoiceField):
+    def validate(self, value):
+        """Validate that the input is a list or tuple."""
+        if self.required and not value:
+            raise ValidationError(self.error_messages['required'], code='required')
+        # Validate that each value in the value list is in self.choices.
+        errors = []
+        for val in value:
+            if not self.valid_value(val):
+                errors.append(ValidationError(
+                    self.error_messages['invalid_choice'],
+                    code='invalid_choice',
+                    params={'value': val},
+                ))
+        if errors:
+            raise ValidationError(
+                errors,
+                code='invalid_choice'
+            )
+
+
+class ApiQueryForm(forms.Form):
+    # study_id is cleaned to the object_id of the chosen Study on output
+    study_id = forms.ModelChoiceField(queryset=Study.objects.all(),
+                                      required=True,
+                                      to_field_name="object_id",
+                                      error_messages={'invalid_choice': "no study with that id was found"})
+
+    end_date = forms.DateField(required=False,
+                               error_messages={'invalid': "end date could not be interpreted as a date. Dates should be"
+                                                          "formatted as 'mm/dd/yyyy' (without quotes)"})
```

Comment:
Let's change this to `YYYY-MM-DD` to match the [Swagger spec for date](https://swagger.io/docs/specification/data-models/data-types/#format).